### DISCUSSION TITLE
Xonsh doesn't automatically use prompt_toolkit on NixOS with GNOME Wayland

### DISCUSSION THREAD

**Discussion by SamLukeYes** [xonsh/xonsh#5675] [Q&A]
On NixOS, with GNOME on Wayland and Xonsh as login shell, Xonsh silently falls back to `dumb` shell type at start, unless `$SHELL_TYPE` is set to `best` or `prompt_toolkit` in an rc file that is loaded, or `--no-env` is passed to Xonsh. As far as I've tested, it doesn't affect Plasma on Wayland or GNOME on Xorg. If `/etc/xonsh/xonshrc`, which is controlled by NixOS modules, doesn't exist at login, the issue also goes away. Though it can be simply worked around, I'm not sure about its root cause. Is there any further investigation I can do?

**Comment by anki-code** [comment]
Hey! As first try to check SHELL_TYPE value by saving it in the first lines of xonshrc. It's the default or upstream value?
As the second try to chech [xonsh code for setting shell type](https://github.com/xonsh/xonsh/blob/27ab7f13316351299cceb3cf7fee580b4a48a946/xonsh/shell.py#L156) and add additional logging for debugging.

**Reply by SamLukeYes** [reply]
Just tried adding `echo $SHELL_TYPE > ~/shell-type` to the first line of `/etc/xonsh/xonshrc`. When I login to GNOME on Wayland, it is `dumb`, while it is `best` on Xorg session.

**Reply by anki-code** [reply]
I think it's because the TERM is empty:
https://github.com/xonsh/xonsh/blob/27ab7f13316351299cceb3cf7fee580b4a48a946/xonsh/shell.py#L172-L173

Check this out as you did for shell type. If it's right feel free to improve this logic.

**Reply by SamLukeYes** [reply]
Oh, I noticed another weird behavior. `echo $SHELL_TYPE > ~/shell-type` always dumps `best` before launching any interactive Xonsh instances, while it becomes `dumb` after launching an interactive Xonsh on Wayland. Also tried adding `echo $TERM > ~/term` before it -- it dumps the string "$TERM" as is before launching interactive Xonsh, and behaves the same between Wayland and Xorg sessions.

**Reply by anki-code** [reply]
> it dumps the string "$TERM"

This means that `$TERM` is not defined i.e. `env | grep TERM` is empty.
